test/method.jl

```diff
@@ -0,0 +1,181 @@
+    haskey(target, :name) && @test target[:name] == get(candidate, :name, nothing)
+    haskey(target, :params) && @test target[:params] == get(candidate, :params, nothing)
+    haskey(target, :args) && @test target[:args] == get(candidate, :args, nothing)
+    haskey(target, :whereparams) &&
+        @test target[:whereparams] == get(candidate, :whereparams, nothing)
+    return nothing
+end
+
+"""
+    only_method(f, [typ])
+
+Return the only method of `f`,
+Similar to `only(methods(f, typ))` in julia 1.4.
+"""
+function only_method(f, typ=Tuple{Vararg{Any}})
+    ms = methods(f, typ)
+    if length(ms) !== 1
+        error("not just one method matches the given types. Found $(length(ms))")
+    end
+    return first(ms)
+end
+
+
+
+@testset "method.jl: signature" begin
+    @testset "Basics" begin
+        @test_signature basic1(x) = 2x
+        @test_signature basic2(x::Int64) = 2x
+        @test_signature basic3(x::Int64, y) = 2x
+        @test_signature basic4() = 2
+    end
+
+    @testset "missing argnames" begin
+        @test_signature ma1(::Int32) = 2x
+        @test_signature ma2(::Int32, ::Bool) = 2x
+        @test_signature ma3(x, ::Int32) = 2x
+    end
+
+    @testset "Whereparams" begin
+        @test_signature f4(x::T, y::T) where T = 2x
+
+        @test_signature f5(x::S, y::T) where {S,T} = 2x
+        @test_signature f6(x::S, y::T) where {T,S} = 2x
+        @test_signature f7(x::S, y::T) where T where S = 2x
+    end
+
+    @testset "Whereparams with constraints" begin
+        @test_signature f8(x::S) where S<:Integer = 2x
+        @test_signature f9(x::S, y::S) where S<:Integer = 2x
+        @test_signature f10(x::S, y::T) where {S<:Integer, T<:Real} = 2x
+        @test_signature f11(x::S, y::Int64) where S<:Integer = 2x
+
+        @test_signature f12(x::S) where {S>:Integer} = 2x
+        @test_signature f13(x::S) where Integer<:S<:Number = 2x
+    end
+
+    @testset "Arg types with type-parameters" begin
+        @test_signature f14(x::Array{Int64, 1}) = 2x
+        @test_signature f15(x::Array{T, 1}) where T = 2x
+
+        @test_signature f16(x::Array{T, 1} where T<:Real) = 2x
+
+        # This is the same method as f16 (other than name), and one displaces the other
+        # but they have different method objects. And different (but equivelent) ASTd
+        # this generates something that should be the same as what `signature(f16)` does
+        # but with a gensym'd variable name
+        f16_alt(x::Array{<:Real, 1}) = 2x
+        f16_alt_sig = signature(first(methods(f16_alt)))
+        @test f16_alt_sig[:name] == :f16_alt
+        @test occursin(  # Hack to deal with gensymed name. Make it a string and use regex
+            r"^Expr\[:\(x::\(Array\{var\"(.*?)\", 1\} where var\"\1\" <: Real\)\)\]$",
+            string(f16_alt_sig[:args])
+        )
+    end
+
+    @testset "anon functions" begin
+        @test_signature (x) -> x  # no args
+        @test_signature (x)->2x
```

Comment:
Does the spacing matter? If so we need a comment
```suggestion
        @test_signature (x) -> 2x
```